modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/partials/minimal.jsp

```diff
@@ -0,0 +1,43 @@
+<%--
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+--%>
+
+<%@ include file="/init.jsp" %>
+
+<form action="<%= fdsSampleDisplayContext.getPortletURL() %>" method="post" name="fm">
+	<input name="<%= Constants.CMD %>" type="hidden" value="<%= Constants.DELETE %>" />
+	<input name="redirect" type="hidden" value="<%= String.valueOf(fdsSampleDisplayContext.getPortletURL()) %>" />
+
+	<frontend-data-set:headless-display
+		additionalProps='<%=
+			HashMapBuilder.<String, Object>put(
+				"greeting", "Hello"
+			).build()
+		%>'
+		apiURL="<%= fdsSampleDisplayContext.getAPIURL() %>"
+		customViewsEnabled="<%= true %>"
+		fdsActionDropdownItems="<%= fdsSampleDisplayContext.getFDSActionDropdownItems() %>"
+		formId="fm"
+		id="<%= FDSSampleFDSNames.FDS_SAMPLES %>"
+		itemsPerPage="<%= 20 %>"
+		namespace="<%= liferayPortletResponse.getNamespace() %>"
+		pageNumber="<%= 1 %>"
+		portletURL="<%= liferayPortletResponse.createRenderURL() %>"
+		propsTransformer="js/SampleFDSPropsTransformer"
+		showManagementBar="<%= false %>"
+		showPagination="<%= false %>"
+		style="fluid"
+	/>
```

Comment:
If we have a showcase of a minimal configuration, we should reduce amount of attributes to minimum. This is how I think API should be, it may not match implementation:
- `additionalProps` - not required, no additional props by default
- `apiURL` - required
- `customViewsEnabled` - not required, disabled by default
- `fdsActionDropdownItems` - not required, no actions by default
- `formId` - not required, `fm` by default
- `id` - required
- `itemsPerPage` - not required, 20 by default
- `namespace` - not required, portlet namespace by default
- `pageNumber` - not required, 1 by default
- `portletURL` - not required. Do we need this? I believe we can get this as an internal mechanism in the tag class.
- `propsTransformer` - not required, no props transformations by default
- `showManagementBar` - not required, true by default
- `showPagination` - not required, true by default
- `style` - not required, `style` by default. Sidenote - do we need this? As far as I know, we plan on having just one set of styles.

To summarize, a showcase of minimal configuration should have only `apiURL` and `id`.

Reply:
Yes, you're right.
This are the mandatory attributes:
```
<frontend-data-set:headless-display
		apiURL="<%= fdsSampleDisplayContext.getAPIURL() %>"
		id="<%= FDSSampleFDSNames.FDS_SAMPLES %>"
		itemsPerPage="<%= 20 %>"
		namespace="<%= liferayPortletResponse.getNamespace() %>"
		pageNumber="<%= 1 %>"
		portletURL="<%= liferayPortletResponse.createRenderURL() %>"
/>
```

Removing `additionalProps` breaks the component. If I understand correctly the component, we should have different `SampleTableFDSView.java` views, each of one with it's proper schema, right?

Reply:
I agree with you that we should make it easier, the less mandatory attributes the better. I wonder where is the right place to pass the default values, the tag?

Reply:
> I agree with you that we should make it easier, the less mandatory attributes the better. I wonder where is the right place to pass the default values, the tag?

Generally, it's better to have default on the frontend component prop, as we are supporting direct usage (not over JSP tag). For some attibutes, tag class default may be a better fit, if they have a backend component. For example, `namespace`.

Reply:
> Removing `additionalProps` breaks the component. If I understand correctly the component, we should have different `SampleTableFDSView.java` views, each of one with it's proper schema, right?

I'm not sure what you mean. `additionalProps` is related to `propsTransformer`. If lack of it breaks the component, that's a bug. Are you sure that it does? If it did, almost all usages FDS in DXP would not work.

Reply:
> * `style` - not required, `style` by default. Sidenote - do we need this? As far as I know, we plan on having just one set of styles.

I think is useful the option to have the FDS boxed or fullwidth, but it's something can be easily done using Clay layout tools/components, so we could get rid of it... 
We can break some things however...

Reply:
> > Removing `additionalProps` breaks the component.

Sorry @markocikos I meant `propsTransformer`...
This: "Removing propsTransformer breaks the component because it expects a content renderer, uhmmm"

> Each FDS instance should have its own view definition, yes. FDS id must match frontend.data.set.name property.
Great, understood